src/main/java/com/github/mc1arke/sonarqube/plugin/almclient/bitbucket/DefaultBitbucketClientFactory.java

```diff
@@ -79,14 +79,15 @@ public BitbucketClient createClient(ProjectAlmSettingDto projectAlmSettingDto, A
                     .orElseThrow(() -> new InvalidConfigurationException(InvalidConfigurationException.Scope.GLOBAL, "URL must be set in configuration"));
             String personalAccessToken = Optional.ofNullable(StringUtils.trimToNull(almSettingDto.getDecryptedPersonalAccessToken(settings.getEncryption())))
                     .orElseThrow(() -> new InvalidConfigurationException(InvalidConfigurationException.Scope.PROJECT, "Personal access token must be set in configuration"));
-            return new BitbucketServerClient(new BitbucketServerConfiguration(almRepo, almSlug, url), objectMapper, createAuthorisingClient(clientBuilder, personalAccessToken));
+            return new BitbucketServerClient(new BitbucketServerConfiguration(almSlug, almRepo, url), objectMapper, createAuthorisingClient(clientBuilder, personalAccessToken));
```

Comment:
can we add a unit test for this constructor and assert values through any exposed getters?

Reply:
There is already a test for this here: https://github.com/mc1arke/sonarqube-community-branch-plugin/blob/675f2bb0ea28876383893113d159315b72078665/src/test/java/com/github/mc1arke/sonarqube/plugin/almclient/bitbucket/BitbucketServerClientUnitTest.java#L229

Arguments were provided to the constructor in the wrong order. Possibly because of confusing variable names, but that's also what they're called in the Sonar API.

Reply:
personally it looks like you are fixing the wrong end, see https://github.com/mc1arke/sonarqube-community-branch-plugin/blob/master/src/main/java/com/github/mc1arke/sonarqube/plugin/almclient/bitbucket/model/server/BitbucketServerConfiguration.java#L27 it expects repo and slug. if you are saying that almrepo should be project and almslug should be project then we should swap the constructor order in https://github.com/mc1arke/sonarqube-community-branch-plugin/blob/f1fd8b4ffb7108d68c858e7dd09e5a64ce893e79/src/main/java/com/github/mc1arke/sonarqube/plugin/almclient/bitbucket/model/BitbucketConfiguration.java#L26

Reply:
Sounds good 👍